tests/publisher/tests_register_name_dispute.py

```diff
@@ -0,0 +1,138 @@
+import responses
+from tests.publisher.endpoint_testing import BaseTestCases
+
+
+class GetRegisterNameDisputePageNotAuth(BaseTestCases.EndpointLoggedOut):
+    def setUp(self):
+        endpoint_url = "/register-name-dispute"
+        super().setUp(snap_name=None, endpoint_url=endpoint_url)
+
+
+class GetRegisterNameDisputePage(BaseTestCases.BaseAppTesting):
+    def setUp(self):
+        super().setUp(
+            snap_name="test-snap",
+            api_url=None,
+            endpoint_url="/register-name-dispute",
+        )
+
+    @responses.activate
+    def test_register_name_dispute_logged_in(self):
+        self._log_in(self.client)
+
+        endpoint_url = "{}?snap-name={}".format(
+            self.endpoint_url, self.snap_name
+        )
+        response = self.client.get(endpoint_url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assert_template_used("publisher/register-name-dispute.html")
+        self.assert_context("snap_name", "test-snap")
+
+    @responses.activate
+    def test_register_name_dispute_redirect_no_snap_name(self):
+        self._log_in(self.client)
+        response = self.client.get(self.endpoint_url)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, "/register-snap")
+
+
+class PostRegisterNameDisputeNotAuth(BaseTestCases.EndpointLoggedOut):
```

Comment:
do you want to keep GET and POST in the same file.

I tried to have a test file per endpoint.
If you prefer that way I don't really mind, it seems easier when searching for a test

Reply:
I don't have any strong feeling about this. I'm happy to follow the current convention.